src/monitors/rocketMonitor.cpp

```diff
@@ -0,0 +1,175 @@
+#include "rocketMonitor.hpp"
+#include <utility>
+#include <utilities/mpiError.hpp>
+#include <utilities/petscError.hpp>
+#include "io/interval/fixedInterval.hpp"
+#include "monitors/logs/stdOut.hpp"
+#include "monitors/logs/log.hpp"
+#include "monitor.hpp"
+
+ablate::monitors::RocketMonitor::RocketMonitor(const std::string nameIn, std::shared_ptr<domain::Region> regionIn, std::shared_ptr<domain::Region> fieldBoundaryIn, std::shared_ptr<eos::EOS> eosIn, const std::shared_ptr<logs::Log>& logIn, const std::shared_ptr<io::interval::Interval>& intervalIn)
+    : name(nameIn), region(std::move(regionIn)), fieldBoundary(std::move(fieldBoundaryIn)), eos(std::move(eosIn)), log(logIn ? logIn : std::make_shared<logs::StdOut>()), interval(intervalIn ? intervalIn : std::make_shared<io::interval::FixedInterval>()) {}
+
+void ablate::monitors::RocketMonitor::Register(std::shared_ptr<solver::Solver> solver) {
+    Monitor::Register(solver);
+//    PetscMPIInt rank;
+//    MPI_Comm_rank(solver->GetSubDomain().GetComm(), &rank) >> checkMpiError;
+//    PetscInt dim;
+//    DMGetDimension(solver->GetSubDomain().GetDM(), &dim) >> checkError;
+}
+
+PetscErrorCode ablate::monitors::RocketMonitor::OutputRocket(TS ts, PetscInt step, PetscReal crtime, Vec u, void* ctx) {
+    PetscFunctionBeginUser;
+    auto monitor = (ablate::monitors::RocketMonitor*)ctx;
+
+    if (monitor->interval->Check(PetscObjectComm((PetscObject)ts), step, crtime)) {
+
+        auto dm = monitor->GetSolver()->GetSubDomain().GetDM(); // get the dm
+        auto solDM = monitor->GetSolver()->GetSubDomain().GetDM(); // get the sol dm
+        auto comm = PetscObjectComm((PetscObject)monitor->GetSolver()->GetSubDomain().GetDM()); // The communicator in which the reduction takes place.
```

Comment:
This works without a problem, but you can also get the comm directly from the subdomain.